scipy/_lib/messagestream.pyx

```diff
@@ -40,7 +40,7 @@ cdef class MessageStream:
         if stdio.remove(self._filename) == 0:
             self._removed = 1
 
-    def __del__(self):
+    def __dealloc__(self):
         self.close()
```

Comment:
From the docs:
> In particular, don’t call any other methods of the object or do anything which might cause the object to be resurrected. It’s best if you stick to just deallocating C data.

This is probably fine as `close()` is `cpdef`'d and thus as a C version is available even if the Python class is in an invalid state